scripts/data/client.py

```diff
@@ -56,41 +64,41 @@ def __str__(self):
         return f"Job(height='{self.height}', step={self.step}, weight='{self.weight}')"
 
 
-# Generator function to create jobs
+def calculate_batch_weight(block_data, mode):
+    if mode == "light":
+        return len(block_data["blocks"])
+    else:
+        return sum(
+            len(tx["inputs"]) + len(tx["outputs"])
+            for block in block_data["blocks"]
+            for tx in block["data"]["transactions"]
+        )
+
+
 def job_generator(start, blocks, step, mode, strategy):
     BASE_DIR.mkdir(exist_ok=True)
     end = start + blocks
-
     height_range, step = (
         ([random.randint(start, end) for _ in range(start, end)], 1)
         if strategy == "random"
         else (range(start, end, step), step)
     )
-
     for height in height_range:
         try:
-            batch_file = BASE_DIR / f"{mode}_{height}_{step}.json"
-
+            batch_file = BASE_DIR / f"{mode}{height}_{step}.json"
             batch_data = generate_data(
                 mode=mode, initial_height=height, num_blocks=step, fast=True
             )
-
-            Path(batch_file).write_text(json.dumps(batch_data, indent=2))
-
+            batch_file.write_text(json.dumps(batch_data, indent=2))
             batch_weight = calculate_batch_weight(batch_data, mode)
             yield Job(height, step, mode, batch_weight, batch_file), batch_weight
         except Exception as e:
             logger.error(f"Error while generating data for: {height}:\n{e}")
 
 
-# Function to process a batch
-def process_batch(job):
-    arguments_file = job.batch_file.as_posix().replace(".json", "-arguments.json")
-
-    with open(arguments_file, "w") as af:
-        af.write(str(format_args(job.batch_file, False, False)))
-
-    result = subprocess.run(
+def run_process(arguments_file):
```

Comment:
It should take whole command as an argument

Reply:
Resolved in the latest fix.